packages/web/src/js/FileUploader.ts

```diff
@@ -198,19 +252,26 @@ class FileUploader extends BaseComponent {
       const itemElement = SelectorEngine.findOne(`li#${name}`);
       this.fileQueue.delete(name);
       itemElement?.remove();
+      this.updateDropZoneVisibility();
     }
   }
 
   onChange(event: Event & { target: HTMLInputElement }) {
     const { target } = event;
     const filesArray = target.files ? [...target.files] : [];
 
-    filesArray.forEach((file) => this.addToQueue(file));
+    let counter = 0;
+    counter += this.fileQueue.size;
+
+    filesArray.forEach((file) => {
```

Comment:
Same comment as I have in React about breaking the loop.

Reply:
This is how it works for the drag and drop as well, I just used it for the normal selection when I discovered the bug

Reply:
done